templates/coredata/swift4.stencil

```diff
@@ -8,18 +8,24 @@ import Foundation
 // swiftlint:disable file_length
 // swiftlint:disable attributes
 // swiftlint:disable vertical_whitespace_closing_braces
-
 // swiftlint:disable identifier_name line_length type_body_length
 {% set accessModifier %}{% if param.publicAccess %}public{% else %}internal{% endif %}{% endset %}
 {% for model in models %}
 {% for name, entity in model.entities where entity.shouldGenerateCode %}
-{% set superclass %}{{ model.entities[entity.superEntity].className | default:"NSManagedObject" }}{% endset %}
-{% set entityClassName %}{{ entity.className | default:entity.name }}{% endset %}
-{% set override %}{% if superclass != "NSManagedObject" %}override {% endif %}{% endset %}
-// MARK: - {{ entityClassName }}
+{% set superclass %}{{ model.entities[entity.superEntity].className|default:"NSManagedObject" }}{% endset %}
+{% set entityClassName %}{{ entity.className|default:"NSManagedObject" }}{% endset %}
+
+// MARK: - {{ entity.name }}
 
+{% if param.generateObjcName %}
+{% if entityClassName|contains:"." %}
+// warning: {{ entityClassName }} cannot be an objc name.
```

Comment:
Didn't look at everything but so you have a casr, among the different fixtures, which contain an entityClassName with a dot so we see the warning comment being generated as expected here in those cases?

Reply:
No, this is just an extra just in case. The generated code won't even work if this happens, you'd get something like:
```swift
class Foo.Bar: NSManagedObject {
}
```

Which won't work. If the user is trying to create a namespaced type, it'd have to be something like:
```swift
extension Foo
  class Bar: NSManagedObject {
  }
}
```
But we have no way of knowing what the user would even be trying to achieve as we have no info about the type `Foo` (or even deeper levels).

Reply:
We could completely disable code generation for these cases, and just generate that comment.

Reply:
We now generate a warning for such types that we cannot represent:
> // Warning: 'We.Cannot.Handle.ImpossibleType' cannot be a valid type name, skipping code generation.